zerver/views/video_calls.py

```diff
@@ -162,7 +162,14 @@ def make_zoom_video_call(
         "settings": {
             "host_video": is_video_call,
             "participant_video": is_video_call,
-        }
+        },
+        # Generate a default password depending on the user settings. This will
+        # result in the password being appended to the returned Join URL.
+        #
+        # If we don't request a password to be set, the waiting room will be
+        # forcibly enabled in Zoom organizations that require some kind of
+        # authentication for all meetings.
+        "default_password": True,
     }
```

Comment:
What's the thinking on generating a default password, rather than just having Zulip generate a password and set it here? That would produce more predictable results.

https://developers.zoom.us/docs/api/rest/reference/zoom-api/methods/#operation/meetingCreate

Reply:
I chose to let Zoom generate a password because we don't know which password rules the Zoom account did configure. Zoom is quite flexible on which rules can be enabled, and I don't see a way to generate a password that meets all possible criteria:

![image](https://github.com/zulip/zulip/assets/2299951/5e499939-fbd8-4c52-afe3-6412b2f6e2a8)

Namely, "only allow numeric passcode" would conflict with most of the rules. That option is actually something people might want to enable in their Zoom organization, because some ways of joining a Zoom meeting (like phone dial in or some conference rooms) only support numeric passcodes AFAIK.

I think it's better to use Zoom's default password behavior (which should be the same as when you generate a new meeting from Zoom itself) rather than trying to generate custom passwords and deal with all of the edge cases.

Reply:
OK. I got worried about the mention of the "PMI" discussion in the documentation for `default_password`, and thus that potentially leaking the user's PMI passcode; but it looks like `use_pmi` defaults to false. 

> Whether to generate a default passcode using the user's settings. This value defaults to false.

> If this value is true and the user has the PMI setting enabled with a passcode, then the user's meetings will use the PMI passcode. It will not use a default passcode.Default: false


And I suppose if you're using a fixed passcode for your PMI meeting, you presumably should understand that anyone you send the link with its passcode in it will be able to join, so it's probably OK regardless.